### ISSUE TITLE
inconsistent behavior depending on where mypy is invoked

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11697]
given:
```
projects/
├─ mypy/
│  ├─ a.py
│  ├─ b.py
├─ other/
│  ├─ a.py
│  ├─ b.py
```

a.py:
```py
from b import b
```

b.py:
```py
b: int = ""
```

```
>mypy projects/mypy/a.py
Success: no issues found in 1 source file
>mypy projects/other/a.py
b.py:1: error: Incompatible types in assignment (expression has type "int", variable has type "str")
Found 1 error in 1 file (checked 1 source file)
```

I have no idea whats going on here, I can consistently reproduce it with these folders(projects/mypy), but not any other folder.

**Comment by sobolevn** [comment]
What if you rename `projects/mypy/a.py` to `projects/my_mypy/a.py`?

**Comment by KotlinIsland** [comment]
Then the error correctly dosplays

**Comment by sobolevn** [comment]
I guess the problem is in name shadowing 🤔

**Comment by KotlinIsland** [comment]
It's highly bizarre, any python files in the mypy directory don't check properly.

And some things do check properly, if I change b.py to `1 + ""` then an error is reported.
